src/diagram/components/dialog/edit-variable-dialog.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from "react";
+import { applySnapshot, getSnapshot } from "mobx-state-tree";
+import { cloneDeep } from "lodash";
+
+import { NormalDialogRow } from "./dialog";
+import { VariableType } from "../../models/variable";
+import { kMaxNameCharacters, kMaxNotesCharacters } from "../../utils/validate";
+
+import "./dialog.scss";
+
+interface IEditVariableDialogContent {
+  name: string;
+  setName: React.Dispatch<React.SetStateAction<string>>;
+  notes: string;
+  setNotes: React.Dispatch<React.SetStateAction<string>>;
+  value: string;
+  setValue: React.Dispatch<React.SetStateAction<string>>;
+  unit: string;
+  setUnit: React.Dispatch<React.SetStateAction<string>>;
+}
```

Comment:
I think it would be better to just pass a Variable model into this component. Then you don't have to duplicate all of fields in the interface here.

It is still probably best to explicitly define the row components for each field since doing that automatically usually is more trouble than it is worth.